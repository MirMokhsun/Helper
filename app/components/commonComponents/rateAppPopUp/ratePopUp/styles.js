import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
    background: {
        flex: 1,
        height: '100%',
        justifyContent: "center",
        alignItems: 'center',
        backgroundColor: 'rgba(23, 37, 51, 0.9)',
    },
    textBtn: {
        fontFamily: "Roboto-Light",
        fontSize: 16,
        textAlign: 'center',
    },
    popUp: {
        padding: 15,
        paddingBottom: 25,
        backgroundColor: '#FFFFFF',
        shadowOffset: { width: 0, height: 4 },
        shadowColor: 'rgba(0, 0, 0, 0.15)',
        shadowRadius: 4,
        elevation: 4,
        shadowOpacity: 1,
        borderRadius: 15,
    },
    tileWrapper: {
        marginBottom: 15
    },
    titleText: {
        lineHeight: 22,
        fontFamily: "Roboto-Light",
        fontSize: 16,
        textAlign: 'center',
        color: '#000756',
    },
    btnWrapper: {
        flexDirection: 'row',
        justifyContent: 'space-evenly'
    },
    btnWhite: {
        width: 118,
        height: 38,
        borderRadius: 20,
        borderWidth: 1,
        borderColor: '#1459D2',
        backgroundColor: '#FFFFFF',
        justifyContent: 'center',
        alignItems: 'center',
    },
    btn: {
        width: "100%",
        height: 20,
        backgroundColor: 'transparent',
        justifyContent: 'center',
        alignItems: 'center',
    },

});