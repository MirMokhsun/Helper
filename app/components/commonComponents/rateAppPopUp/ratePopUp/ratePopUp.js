import React from 'react';
import { Text, View, TouchableWithoutFeedback, TouchableOpacity, BackHandler, Keyboard } from 'react-native';
import { styles } from './styles';

export class RatePopup extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            height: this.height,
        };
    }

    componentDidMount = () => {
        BackHandler.addEventListener('hardwareBackPressPopup', this.hardwareBackPressPopup);
        this._keyboardWillShowSubscription = Keyboard.addListener('keyboardDidShow', (e) => this._keyboardWillShow(e));
        this._keyboardWillHideSubscription = Keyboard.addListener('keyboardDidHide', (e) => this._keyboardWillHide(e));
    }

    UNSAFE_componentWillUnmount = () => {
        BackHandler.removeEventListener('hardwareBackPressPopup', this.hardwareBackPressPopup);
        this._keyboardWillShowSubscription.remove();
        this._keyboardWillHideSubscription.remove();
    }


    _keyboardWillShow(e) {
        this.setState({ height: this.height - e.endCoordinates.height });
    }

    _keyboardWillHide(e) {
        this.setState({ height: this.height });
    }

    hardwareBackPressPopup = () => {
        const { backgroundFunction } = this.props;
        backgroundFunction();
    }

    getHeightBackgroundPopUp = (event) => {
        const { height } = event.nativeEvent.layout;
        this.height = height;
    }

    render() {
        const {
            width,              // ширина попапа, обязательна
            backgroundFunction, // функция которая вешаеться на пространство за попапом, закрывающая
            topComponent,       // дополнительный компонент, елси его нет, то можно не чего не вводить
            title,              // основной текст, необязателен
            textBtn,            // текст первой кнопки
            functionBtn,        // функция первой кнопки
            textBtnWhite,       // текст второй кнопки, если ее нет то, false
            functionBtnWhite    // функция второй кнопки, если ее нет то, false
        } = this.props;
        const { height } = this.state;
        return (
            <TouchableWithoutFeedback testID="closePopupID" accessibilityLabel="closePopupID" onPress={backgroundFunction}>
                <View style={[styles.background, { maxHeight: height }]} onLayout={this.getHeightBackgroundPopUp} >
                    <TouchableWithoutFeedback onPress={() => {
                        Keyboard.dismiss();
                    }} >
                        <View style={[styles.popUp, { width }]}>
                            {/* в блоке ниже будет свг или дополнительный текст */}
                            {topComponent || null}
                            {/* ниже тайтл */}
                            {title ? (
                                <View style={styles.tileWrapper}>
                                    <Text style={styles.titleText}>{title}</Text>
                                </View>
                            ) : null}
                            <View style={styles.btnWrapper}>
                                {/* Белая кнопка */}
                                {textBtnWhite ? (
                                    <TouchableOpacity onPress={functionBtnWhite} style={styles.btnWhite} testID="leftBtnPopupID" accessibilityLabel="leftBtnPopupID"    >
                                        <Text style={[styles.textBtn, { color: '#1459D2' }]}>{textBtnWhite}</Text>
                                    </TouchableOpacity>
                                ) : null}
                                {/* Синяя кнопка */}
                                {textBtn ? (
                                    <TouchableOpacity onPress={functionBtn} style={styles.btn} testID="rightBtnPopupID" accessibilityLabel="rightBtnPopupID" >
                                        <Text style={[styles.textBtn, { color: '#1459D2' }]}>{textBtn}</Text>
                                    </TouchableOpacity>
                                ) : null}
                                {/*  */}
                            </View>
                        </View>
                    </TouchableWithoutFeedback>
                </View>
            </TouchableWithoutFeedback >
        );
    }
}
