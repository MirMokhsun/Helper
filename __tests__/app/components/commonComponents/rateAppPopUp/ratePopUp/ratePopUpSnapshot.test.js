import 'react-native';
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { RatePopup } from '../../../../../../app/components/commonComponents/rateAppPopUp/ratePopUp/ratePopUp';

it('create snapshot RatePopup', () => {
    const renderer = new ShallowRenderer()
    let listPicker = renderer.render(<RatePopup />);
    expect(listPicker).toMatchSnapshot();
});