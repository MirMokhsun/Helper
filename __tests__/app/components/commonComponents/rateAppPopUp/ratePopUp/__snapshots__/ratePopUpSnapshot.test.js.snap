// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create snapshot RatePopup 1`] = `
<TouchableWithoutFeedback
  accessibilityLabel="closePopupID"
  testID="closePopupID"
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(23, 37, 51, 0.9)",
          "flex": 1,
          "height": "100%",
          "justifyContent": "center",
        },
        Object {
          "maxHeight": undefined,
        },
      ]
    }
  >
    <TouchableWithoutFeedback
      onPress={[Function]}
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 15,
              "elevation": 4,
              "padding": 15,
              "paddingBottom": 25,
              "shadowColor": "rgba(0, 0, 0, 0.15)",
              "shadowOffset": Object {
                "height": 4,
                "width": 0,
              },
              "shadowOpacity": 1,
              "shadowRadius": 4,
            },
            Object {
              "width": undefined,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-evenly",
            }
          }
        />
      </View>
    </TouchableWithoutFeedback>
  </View>
</TouchableWithoutFeedback>
`;
